dll/win32/srclient/srclient_main.c

```diff
@@ -29,40 +31,79 @@
 
 BOOL
 WINAPI
-SRSetRestorePointA(PRESTOREPOINTINFOA pRestorePtSpec, PSTATEMGRSTATUS pStateMgrStatus)
+SRSetRestorePointA(PRESTOREPOINTINFOA pRestorePtSpec, PSTATEMGRSTATUS pSMgrStatus)
 {
-    RESTOREPOINTINFOW RPInfoW;
-
-    DPRINT1("SRSetRestorePointA is unimplemented\n");
-
-    if (!pRestorePtSpec)
-        return FALSE;
-
-    RPInfoW.dwEventType = pRestorePtSpec->dwEventType;
-    RPInfoW.dwRestorePtType = pRestorePtSpec->dwRestorePtType;
-    RPInfoW.llSequenceNumber = pRestorePtSpec->llSequenceNumber;
-
-    if (!MultiByteToWideChar( CP_ACP, 0, pRestorePtSpec->szDescription, -1, RPInfoW.szDescription, MAX_DESC_W))
-        return FALSE;
-
-    return SRSetRestorePointW(&RPInfoW, pStateMgrStatus);
+    HMODULE hModule;
+    PSRSRPA pSRSRPA;
+    BOOL bStatus = FALSE;
+    LPCWSTR lpLibFileName = L"srclient.dll";
+    LPCSTR lpProcName = "SRSetRestorePointA";
+
+    hModule = LoadLibraryW(lpLibFileName);
+    if (hModule)
+    {
+        pSRSRPA = (PSRSRPA)GetProcAddress(hModule, lpProcName);
+        if (pSRSRPA)
+        {
+            bStatus = pSRSRPA(pRestorePtSpec, pSMgrStatus);
+        }
+        else
+        {
+            if (pSMgrStatus)
+            {
+                pSMgrStatus->nStatus = ERROR_CALL_NOT_IMPLEMENTED;
+            }
+        }
+        FreeLibrary(hModule);
+    }
+    else
+    {
+        if (pSMgrStatus)
+        {
+            pSMgrStatus->nStatus = ERROR_CALL_NOT_IMPLEMENTED;
+        }
+    }
+
+    return bStatus;
 }
 
 
 BOOL
 WINAPI
-SRSetRestorePointW(PRESTOREPOINTINFOW pRestorePtSpec, PSTATEMGRSTATUS pStateMgrStatus)
+SRSetRestorePointW(PRESTOREPOINTINFOW pRestorePtSpec, PSTATEMGRSTATUS pSMgrStatus)
```

Comment:
Unnecessary variable rename.

Reply:
But why I need to use the different variables than in documentation? Is it easier to use `pStateMgrStatus` in the whole function instead of changing the variable?

Reply:
Following MSDN documentation for parameter naming is not required, and the old name seems clearer.